packages/web/src/components/gcds-pagination/gcds-pagination.tsx

```diff
@@ -0,0 +1,265 @@
+
+@Component({
+  tag: 'gcds-pagination',
+  styleUrl: 'gcds-pagination.css',
+  shadow: false,
+  scoped: true
+})
+export class GcdsPagination {
+  @Element() el: HTMLElement;
+
+  private lang: string;
+  private listitems = [];
+  private mobilePrevNext = [];
+
+  /*
+   * Props
+   */
+
+  /**
+   * Navigation element label
+   */
+  @Prop() display: "list" | "simple" = "list";
+
+  /**
+   * Navigation element label
+   */
+  @Prop() label!: string;
+
+  /**
+   * Simple display - href for previous link
+   */
+  @Prop() previousHref: string;
+
+  /**
+   * Simple display - label for previous link
+   */
+  @Prop({ reflect: true }) previousLabel: string;
+
+  /**
+   * Simple display - href for next link
+   */
+  @Prop() nextHref: string;
+
+  /**
+   * Simple display - lable for next link
+   */
+  @Prop({ reflect: true }) nextLabel: string;
+
+  /**
+   * List display - Total number of pages
+   */
+  @Prop() totalPages: number;
+
+  /**
+   * List display - Current page number
+   */
+  @Prop({ reflect: true }) currentPage: number;
+
+  @Watch('currentPage')
+  watchCurrentPage(newValue) {
+    this.currentStep = newValue;
+  }
+
+  /**
+   * List display - URL object to create query strings and fragment on links
+   */
+   @Prop() url: Object;
+
+  /**
+   * Function to fire when pageChange event is called
+   */
+  @Prop() pageChangeHandler: Function;
+
+  @State() currentStep: number;
+
+  /*
+   * Events
+   */
+
+  /**
+   * Update value based on user input.
+   */
+  @Event() gcdsPageChange: EventEmitter<void>;
+
+  private onPageChange = (e) => {
+    if (this.pageChangeHandler) {
+      this.pageChangeHandler(e);
+    }
+
+    this.gcdsPageChange.emit();
+  }
+
+  /**
+   * Function to constuct <li> and <a> tags for display="list" pagination
+   */
+  private configurePaginationStep(page: number, end?: "next" | "previous" | null, mobile?: boolean) {
+
+    let linkAttrs = {
+      onClick: (e) => this.onPageChange(e),
+      href: this.url ? constructHref(this.el, page, end) : "javascript:void(0)",
+      "aria-label": !end ?
+        I18N[this.lang].pageNumberOf.replace('{#}', page).replace('{total}', this.totalPages).replace('{label}', this.label)
+      :
+        end == "next" ?
+          `${I18N[this.lang].nextPage}: ${I18N[this.lang].pageNumberOf.replace('{#}', ++page).replace('{total}', this.totalPages).replace('{label}', this.label)}`
+        :
+          `${I18N[this.lang].previousPage}: ${I18N[this.lang].pageNumberOf.replace('{#}', --page).replace('{total}', this.totalPages).replace('{label}', this.label)}`,
+    }
+
+    if (page == this.currentPage && !end) {
+      linkAttrs['aria-current'] = "page";
+    }
+
+    if (end == "next") {
+      return(<li><a  {...linkAttrs} class={!mobile ? "gcds-pagination-end-button" : "gcds-pagination-end-button-mobile"}>{I18N[this.lang].next}<gcds-icon margin-left="spacing-200" name="arrow-right"></gcds-icon></a></li>);
```

Comment:
Can you format this a bit more so that it's easier to read? Something like:
`
return(
        <li>
          <a
            {...linkAttrs}
            class={!mobile ? "gcds-pagination-end-button" : "gcds-pagination-end-button-mobile"}
          >
            {I18N[this.lang].next}
            <gcds-icon margin-left="spacing-200" name="arrow-right"></gcds-icon>
          </a>
        </li>
      );
`